content/docs/reference/design-docs/conflict-detection/index.en.md

```diff
@@ -9,131 +9,210 @@ weight: 45
-  It can be done by finding when the driver would be constrained by signaling, and using that minus an allowance as the deadline.
-- for each zone in each route, simulate when it would be released, and thus not required anymore
-- use the route calling deadline as the requirement start time for each zone, and the release time as the requirement end. It is a shareable requirement.
-  Switch requirements use the same time bounds.
+- Compute the set deadline using signaling simulation. The set deadline is the point in time at which the train would be slowed down if the route were not set.
+- For each zone in each route, simulate when it would be released, and thus not required anymore.
 
 {{% pageinfo color="info" %}}
-Implementing route overlaps alternatives would require a richer model for requirements, such as a hierarchy and alternatives
+Route overlaps are not yet supported.
 {{% /pageinfo %}}
 
-### Spacing requirements
+### Requirement compatibility rules
+
+Requirement compatibility is evaluated for all `RouteZoneRequirement`s, grouped by zone. Requirements A and B, ordered such that `A.set_deadline <= B.set_deadline`, are compatible if and only if either:
+
+ - their active time span does not overlap, such that `A.release_time <= (B.set_deadline - activation_time)`, where the activation time is the delay required to reconfigure from `A.switches` to `B.switches`.
+ - `(A.entry_det, A.exit_det, A.switches) == (B.entry_det, B.exit_det, B.switches)`
+
+
+## Spacing conflicts
+
+### Context
 
-Trains can also be slowed down by catching up with another train.
-When this happens, signaling forces the following train to slow down,
-or even stop if the train being followed does not clear the way fast enough.
+Even if interlocking mitigates some of the risks associated with operating trains, a major one is left out: head to tail collisions, caused by insufficient spacing.
 
-These slowdowns are conflicts, and thus need to be avoided by emitting sufficient requirements.
+This responsibility is handled by signaling, which conveys both interlocking and spacing constraints.
 
-**At any time, exclusive requirements are emitted for zones which if occupied, would trigger a slowdown**
+Signaling helps trains slow down until the end of their movement authority, which is either:
 
-These requirements are emitted as follows:
- - the route calling timeline is followed
- - every time the driver sees a signal:
-   - start by assuming routes reserved for routing should be reserved for spacing as well
-   - until the driver sees a signal which would cause a slowdown:
-     - attempt to make the last reserved zone occupied by a virtual train
-     - if the signal now causes a slowdown, the currently reserved zones are correct, continue to the next signal
-     - if not, unreserve the last zone of the path
+ - behind the tail of the next train
+ - at the end of the last route set for this train
```

Comment:
Assuming the first option is for moving block, there's a missing case with the end of the last free block

Reply:
there must be a misunderstanding: the movement authority always ends either at the end of the last set route, or somewhere behind a train which also uses the same set route.
what do you mean by free block?

Reply:
The misunderstanding was in the "somewhere behind a train" part. I understood this sentence as "the MA ends precisely at the tail of the next train".

But actually it is pretty clear, I just read it wrong.